airflow/cli/commands/task_command.py

```diff
@@ -126,10 +126,12 @@ def _get_dag_run(
         dag_run = DagRun(dag.dag_id, run_id=exec_date_or_run_id, execution_date=dag_run_execution_date)
         return dag_run, True
     elif create_if_necessary == "db":
+        data_interval = dag.infer_automated_data_interval(logical_date=dag_run_execution_date)
```

Comment:
This would not always work since infer_automated_data_interval` is only designed to provide backward compatibility for DAG runs that are created before the data interval concept was introduced, and a DAG using a custom timetable would not work correctly with this method. I think `infer_manual_data_interval` is probably a better choice; it is not technically correct (the call below created a scheduled run), but the exact value of the data interval does not actually matter much here anyway since the run is only supposed to be a stub.

Reply:
yes, since `run_type` is not passed here this just calls the same part as underlying code to have same behavior. It's infer_manual_data_interval in tests since they pass run_type as manual. The dagrun is also deleted in the end so I guess it's ok.

https://github.com/apache/airflow/blob/ea55626d79fdbd96b6d5f371883ac1df2a6313ec/airflow/models/dag.py#L2582-L2585

Reply:
I took the liberty to change this myself.